lib/range.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+const rangeParser = require('parse-numeric-range');
+
+function parseFileSelections(files) {
+	const ranges = new Map();
+	const ignored = [];
+	const filenames = [];
+
+	for (const file of files) {
+		const [actualFilename, lines] = parseFileSelection(file);
+
+		if (!ranges.has(actualFilename)) {
+			ranges.set(actualFilename, []);
+			filenames.push(actualFilename);
+		}
+
+		if (lines.length === 0 && ranges.get(actualFilename).length > 0) {
```

Comment:
Let's move this up, and change the condition to `lines.length === 0 && ranges.has(actualFilename)`.